torch/_inductor/utils.py

```diff
@@ -2272,7 +2272,9 @@ def is_collective(
     from . import ir
 
     return (
-        type(node) == ir._CollectiveKernel and (op is None or node.op_overload is op)
+        isinstance(node, ir._CollectiveKernel)
+        and not isinstance(node, ir._WaitKernel)
```

Comment:
From the definition of `ir._WaitKernel` which is a subclass of `ir._CollectiveKernel`, why it will be excluded here?

Reply:
Because `ir._WaitKernel` does not have `constant_args` like other collective kernels.

The function `is_collective` is used here https://github.com/pytorch/pytorch/blob/main/test/inductor/test_snode_runtime.py#L244. If the check is `True`, it will finally call `_get_group_size_by_name(node.constant_args[-1])` https://github.com/pytorch/pytorch/blob/main/torch/_inductor/comm_analysis.py#L73, which raises an error for `ir._WaitKernel`.
Ditto for the change of `get_collective_group_size` in `common_analysis.py`.

Reply:
Then shouldn't get_collective_group_size handle handles _WaitKernel differently? (disclaimer: I am not sure if _WaitKernel has a concept of group size)

Reply:
`_WaitKernel` is a special `_CollectiveKernel`, which does not have `group_size` (generate from `group_name`) like others.

For example, we compare the schemas of `wait_tensor` and `all_reduce`:
`"wait_tensor(Tensor tensor) -> Tensor"`
`"all_reduce(Tensor input, str reduce_op, str group_name) -> Tensor"`